src/sql/base/browser/ui/infoButton/infoButton.ts

```diff
@@ -0,0 +1,178 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Button as sqlButton } from 'sql/base/browser/ui/button/button';
+import * as DOM from 'vs/base/browser/dom';
+import { IButtonOptions } from 'vs/base/browser/ui/button/button';
+import { URI } from 'vs/base/common/uri';
+
+type IUserFriendlyIcon = string | URI | { light: string | URI; dark: string | URI };
+
+export interface IInfoButtonOptions extends IButtonOptions {
+	description: string,
```

Comment:
If these are not guaranteed to be here, place a ? at the end of the field names, basically if they're optional put the `?`

Reply:
Well, they ARE guaranteed in the sense that they are supplied by dashboardWidget. If I wanted to make this more flexible, these would essentially all have to be optional AND the element creation would have to be suppressed for any part of the layout lacking properties from the widget. I think that's beyond what I'm trying to do here BUT wise considering the possibility of reuse. Maybe open a new PR to work that direction after the extension is deployed?

Reply:
I think if you turn on strict compiles for this it will start complaining that these could be undefined - not a big deal but eventually we're going to end up having to do it

Reply:
Can I just initialize each with base values: false, '', 0 ?

Reply:
Alan just showed me a way of declaring and using the vars without relying on initializing.